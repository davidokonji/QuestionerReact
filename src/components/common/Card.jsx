import React from 'react';
import moment from 'moment';
import { Link } from 'react-router-dom';

const Card = ({
  id, images, title, location, tags, happeningOn
}) => {
  const month = moment(happeningOn).format('MMM');
  const day = moment(happeningOn).format('D');
  const date = moment(happeningOn).format('dddd, MMMM Do YYYY');
  return (
    <div
      className='card col-md-3 col-lg-3 my-3 shadow-sm p-0 mx-3'
      role='button'
      tabIndex='-1'
      onKeyDown={null}
    >
      <div className='card-body p-0'>
        <div
          className='card-img'
        >
          <img src={images} alt='' className='card-img-top card-img' />
        </div>
        <div className='row col-12 h-auto'>
          <div className='col-2 col-md-2 d-flex flex-column side-card p-0'>
            <span className='mt-5 side-card-value mx-auto'>
              {month}
            </span>
            <span className='mt-2 side-card-value mx-auto'>
              {day}
            </span>
          </div>
          <div className='col-10 col-md-10'>
            <div className='col-12 my-3'>
              <Link to={`/meetup/${id}`} className='stretched-link card-link'>
                {title}
              </Link>
            </div>
            <div className='col-12 my-3'>
              {date}
            </div>
            <div className='col-12 my-3'>
              {location}
            </div>
            <div className='col-12 my-3'>
              <span className=''>
                {tags}
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Card;
