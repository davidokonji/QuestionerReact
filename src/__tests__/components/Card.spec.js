import React from 'react';
import { shallow } from 'enzyme';
import Card from '../../components/common/Card';


const data = {
  id: '1',
  title: 'this is a test',
  images: '',
  location: 'lagos',
  happeningOn: '',
  tags: ['test']
};

describe('Card component', () => {
  it('should render card component', () => {
    const wrapper = shallow(<Card />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render card component with props', () => {
    const wrapper = shallow(<Card {...data} />);

    expect(wrapper).toMatchSnapshot();
  });
});
