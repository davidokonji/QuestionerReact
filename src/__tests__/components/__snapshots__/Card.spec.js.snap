// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render card component 1`] = `
<div
  className="card col-md-3 col-lg-3 my-3 shadow-sm p-0 mx-3"
  onKeyDown={null}
  role="button"
  tabIndex="-1"
>
  <div
    className="card-body p-0"
  >
    <div
      className="card-img"
    >
      <img
        alt=""
        className="card-img-top card-img"
      />
    </div>
    <div
      className="row col-12 h-auto"
    >
      <div
        className="col-2 col-md-2 d-flex flex-column side-card p-0"
      >
        <span
          className="mt-5 side-card-value mx-auto"
        >
          May
        </span>
        <span
          className="mt-2 side-card-value mx-auto"
        >
          18
        </span>
      </div>
      <div
        className="col-10 col-md-10"
      >
        <div
          className="col-12 my-3"
        >
          <Link
            className="stretched-link card-link"
            to="/meetup/undefined"
          />
        </div>
        <div
          className="col-12 my-3"
        >
          Saturday, May 18th 2019
        </div>
        <div
          className="col-12 my-3"
        />
        <div
          className="col-12 my-3"
        >
          <span
            className=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Card component should render card component with props 1`] = `
<div
  className="card col-md-3 col-lg-3 my-3 shadow-sm p-0 mx-3"
  onKeyDown={null}
  role="button"
  tabIndex="-1"
>
  <div
    className="card-body p-0"
  >
    <div
      className="card-img"
    >
      <img
        alt=""
        className="card-img-top card-img"
        src=""
      />
    </div>
    <div
      className="row col-12 h-auto"
    >
      <div
        className="col-2 col-md-2 d-flex flex-column side-card p-0"
      >
        <span
          className="mt-5 side-card-value mx-auto"
        >
          Invalid date
        </span>
        <span
          className="mt-2 side-card-value mx-auto"
        >
          Invalid date
        </span>
      </div>
      <div
        className="col-10 col-md-10"
      >
        <div
          className="col-12 my-3"
        >
          <Link
            className="stretched-link card-link"
            to="/meetup/1"
          >
            this is a test
          </Link>
        </div>
        <div
          className="col-12 my-3"
        >
          Invalid date
        </div>
        <div
          className="col-12 my-3"
        >
          lagos
        </div>
        <div
          className="col-12 my-3"
        >
          <span
            className=""
          >
            test
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
